File: Src/Icm.ContextConsole/Application/IApplicationContext.vb
Imports Icm.Localization


Public Interface IApplicationContext

    ReadOnly Property CurrentContextNode As ITreeNode(Of IContext)
    ReadOnly Property RootContextNode As ITreeNode(Of IContext)

    Function GetAllContexts() As IEnumerable(Of IContext)

    ReadOnly Property TokenQueue As Queue(Of String)

    ReadOnly Property Interactor As IInteractor

    ReadOnly Property InternalLocRepo As ILocalizationRepository
    ReadOnly Property ExternalLocRepo As ILocalizationRepository

End Interface

File: Src/Icm.ContextConsole/Application/IApplicationContext.cs
using System.Collections.Generic;
using Icm.Localization;
using Icm.Tree;

public interface IApplicationContext
{

	ITreeNode<IContext> CurrentContextNode { get; }

	ITreeNode<IContext> RootContextNode { get; }
	IEnumerable<IContext> GetAllContexts();


	Queue<string> TokenQueue { get; }

	IInteractor Interactor { get; }
	ILocalizationRepository InternalLocRepo { get; }

	ILocalizationRepository ExternalLocRepo { get; }
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
